fix admin device page and improve UX

LizMintOS opened the pull request:
## Ticket Reference

Issue #279 

## Purpose of Changes

My ticket is to fix the admin device page. There are bugs with the edit device modal and the add device button/modal 1. doesn't work and 2. isn't needed. Devices are added automatically to the frontend when detected by the webhook. However, this currently doesn't work as intended and needs to be fixed.

## Technical Solution

Fixed device property conditional rendering i.e. if device has no block/room, it shows "unassigned" in those columns.

Fixed seed data to match schema tables, display data correctly, and added more for edge cases.

Moved Popup component functions to appropriate hooks.

Fixed popup UI so that it includes more inputs for device info and is styled for better readability.

Improved Add/Edit Device modal UI using the following as examples:
<div align="center">
<img src="https://github.com/user-attachments/assets/327ab339-fc95-4ca8-9341-beef53868044" width="500"/>
</div>

Removed "Add New Device" button.

# Known Bugs
Submitting new changes for device location is not updated unless the user refreshes.

## For Testing/Review
Please tick off if the following are met (where applicable):

- [ ] Meets design requirements
- [ ] Meets coding standards
- [ ] Functions as intended
- [ ] No unused code
- [ ] No/minimal bugs
- [ ] Decent error handling
- [ ] Meets acceptance criteria

## Acceptance Criteria

Button should open a modal popup that displays success/error messages appropriately

On submit, the user should see their new device added to the device list and the correct block

Users should be able to press a clearly labelled button to edit their device and see a modal show up

Users should be able to edit the device location information with adequate response feedback

Users should be able to see the updated device information on the devices table

Module should appear on TTN and in DB

Client-facing users should be able to see new module gauge and module page correctly

Developers should see the module correctly sending live data to the TTN platform

mahoneybj reviewed (changes requested):
Doing excellent work here Liz, thank you for fixing up the seeding files and fixing issues with our backend! Great use of components in the front end. I have a few suggestions and bugs I have found.

Bugs:
- (You have already noted) page not reacting to updated changes like when changing room or block, manual reload required.
- I can't seem to unassign the block to a sensor. E.g current block is D-block, when I try to change it to unassigned.
```
Invalid `prisma.device.update()` invocation:

An operation failed because it depends on one or more records that were required but not found. No 'Block' record(s) (needed to inline the relation on 'Device' record(s)) was found for a nested connect on one-to-many relation 'BlockToDevice'.
```
- Only 1 sensor can be unassigned a room number:
![image](https://github.com/user-attachments/assets/fc38450c-7d6d-48e9-af24-aef2d325e6e7)

Suggestions:

- Adding some user feedback could be handy, like a success message.
-  Adding some comments to your hook methods explaining main points.

However other than that its great work on what was quite a big issue especially with the backend bugs!

LizMintOS commented:
Thanks for the review. I just got stuck and frustrated, so thanks for testing for me.

About the unassigned issue, I knew the schema only allowed unique room numbers but I didn't change it as I'm not confident in making migration changes that might affect the live service.

With changing the block to unassigned, I believe you need to also change the room number to unassigned. I'll make it so room number automatically changes to unassigned there. Apart from that, I'm not sure I understand the problem here so would you be free to show me in person or describe the problem further?

@mahoneybj

LizMintOS commented:
Hey @mahoneybj I just finished those requested changes. I've fixed the bugs and made a lot of improvements to design and functionality. Let me know what you think. I made the following changes:

- Text styling in columns so important details are semibold
- Put device eui and ID/name in the same column with ID/name in bold as it makes more sense and adheres to best practices
- Adjusted the dateTimeFunction to accurately calculate and display last seen status with additional details in the bottom text
- Added loading, success, and error handling with user feedback
- Fixed the manual refresh issue
- Adjusted table padding
- Can now assign devices to same room and block

mahoneybj reviewed (changes requested):
Looking great Liz, love the design enhancements! 

I have 1 bug and a couple of potential suggestions. 

The device popup isn't reactive to screen sizes, if I shrink my screen size down a bit it cuts off content and buttons, see screenshots below:
![Screenshot From 2025-05-13 14-51-02](https://github.com/user-attachments/assets/eee41701-ddfa-4aa0-812e-d89e9cdeb5f5)
![Screenshot From 2025-05-13 14-38-46](https://github.com/user-attachments/assets/b432353f-28db-46ae-bedc-5e9b8f1a8a55)

Suggestions:

- Catch prisma error message with simple user friendly error message when trying to set a device to a room already used by another device. See screenshot:

![Screenshot From 2025-05-13 14-43-14](https://github.com/user-attachments/assets/117dc93e-c203-4de7-bdf9-2cfd12f0eb23)

- Remove unused/commented out code. 
  - ```AdminPanel.jsx (line 2, line 23)```
  - ```AdminTable.jsx (line 1 react import)```
  - ```AdminTableBody.jsx (line 1 react import)```
  - ```DevicePopUp.jsx (line 1 react import)```

Other than that it is ready to go! Let me know when you have fixed the device popup modal and if you get a chance to implement my suggestions mainly the unused code and I will be happy to quickly review and approve.

mahoneybj reviewed (approved):
Those changes are perfect, all working well and looks great!

Good job
